src/app/services/breadcrumbs/breadcrumbs.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, Data, NavigationEnd, Router } from '@angular/router';
+
+import { Observable, Subject } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+
+import { BreadcrumbsStep } from '@app/interfaces/helpers/breadcrumbs-step.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BreadcrumbsService {
+  private showBreadcrumbs$$ = new Subject<boolean>();
+  private breadcrumbsSteps$$ = new Subject<BreadcrumbsStep[]>();
+
+  constructor(private router: Router, private activatedRoute: ActivatedRoute) {
+    this.subscribeOnRouterEvents();
+  }
+
+  get showBreadcrumbs$(): Observable<boolean> {
+    return this.showBreadcrumbs$$.asObservable();
+  }
+
+  get breadcrumbsSteps$(): Observable<BreadcrumbsStep[]> {
+    return this.breadcrumbsSteps$$.asObservable();
+  }
+
+  private subscribeOnRouterEvents() {
+    /**
+     * There is no straightforward way to get route static data outside of <router-outlet></router-outlet> component
+     * The workaround below helps to overcome this issue
```

Comment:
I'm although faced this challenge, it wasn't easy to find, but the way still exist. I used **Router** provider for this purpose.
Using **Router.routerState.snapshot.root** - returns **ActivatedRouteSnapshot** of the root element, from this root element we can get route static **data** and **firstChild**  properties.
**firstChild** - contain child **ActivatedRouteSnapshot** with corresponding static **data** and **firstChild**
So just moving by firstChild chain we can access all needed data